### PR TITLE
Document Project API server.

### PR DESCRIPTION
**Pull request by areusch**
@gromero @mehrdadh @guberti 

#8595

### CODE REVIEW

File: docs/arch/microtvm_project_api.rst [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+.. _microtvm_project_api:
+
+microTVM Project API
+====================
+
+About microTVM Project API
+--------------------------
+
+The microTVM Project API allows TVM to automatically run models on
+unconventional platforms. It allows platforms to define a standard function to
```
Comment (leandron) [inline docs/arch/microtvm_project_api.rst:27]: I think the use of `unconventional` here makes this sentence a bit vague, perhaps we could just say _embedded_ (?), given the paragraph goes towards producing a _firmware_, which then makes everything more specific.

**Review by leandron — COMMENTED** [review]
LGTM in general, just one comment in order to make the wording more specific.

File: docs/arch/microtvm_project_api.rst (hunk 1 above)
Comment (areusch) [reply docs/arch/microtvm_project_api.rst]: added "unconventional or embedded," maybe that helps. agree unconventional is a bit vague but given the example make flow i do think it's a bit limiting. the concept i want to convey here is "deploy in scenarios where you have to build code using an external build tool."

File: docs/arch/microtvm_project_api.rst [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+5. Verify the API server is version-compatible with TVM, plus read properties
+   of the implementation, by sending ``server_info_query`` command.
+6. Build and flash the projec by sending commands ``build`` and ``flash`` to the
+   API server.
+7. Communicate with the target. Send command ``open_transport`` followed by
+   commands ``write_transport`` and ``read_transport`` to write and read from
+   e.g. a serial port attached to the target. Upon completion,
+   ``close_transport`` is sent.
+8. Terminate Project API server.
+
```
Comment (leandron) [inline docs/arch/microtvm_project_api.rst:93]:
```suggestion
4. Terminate the Template Project API server.
5. Launch Project API server in Generated Project.
6. Verify the API server is version-compatible with TVM, plus read properties
   of the implementation, by sending ``server_info_query`` command.
7. Build and flash the projec by sending commands ``build`` and ``flash`` to the
   API server.
8. Communicate with the target. Send command ``open_transport`` followed by
   commands ``write_transport`` and ``read_transport`` to write and read from
   e.g. a serial port attached to the target. Upon completion,
   ``close_transport`` is sent.
9. Terminate Project API server.

```

**Review by leandron — APPROVED** [review]
Just another very minor thing, now I finished.

File: docs/arch/microtvm_project_api.rst (hunk 1 above)
Comment (leandron) [reply docs/arch/microtvm_project_api.rst]: ack. It is more clear now.

**Pull request merged**